diff --git a/build/transport/WebSocketTransport.cjs b/build/transport/WebSocketTransport.cjs
index 572d90b626084b0c4f6b24998b2b10a4a8b2e8af..7371efb7328eda5f18d532c05fcf2ea59261e665 100644
--- a/build/transport/WebSocketTransport.cjs
+++ b/build/transport/WebSocketTransport.cjs
@@ -25,11 +25,13 @@ class WebSocketTransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     connect(url, headers) {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-        }
-        catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
diff --git a/build/transport/WebSocketTransport.mjs b/build/transport/WebSocketTransport.mjs
index 0a9de3b89f6e75096ea3297cf4847b2dca23df41..0bff677b2a268832b99eccd696adb3fb8a4473e4 100644
--- a/build/transport/WebSocketTransport.mjs
+++ b/build/transport/WebSocketTransport.mjs
@@ -26,11 +26,13 @@ class WebSocketTransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     connect(url, headers) {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-        }
-        catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
diff --git a/src/transport/WebSocketTransport.ts b/src/transport/WebSocketTransport.ts
index 83a4782d78e86667142be24eb59a139f7b95a33b..e8f6cceed396492ef37f01ca587c6cc19e9929a2 100644
--- a/src/transport/WebSocketTransport.ts
+++ b/src/transport/WebSocketTransport.ts
@@ -27,11 +27,13 @@ export class WebSocketTransport implements ITransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     public connect(url: string, headers?: any): void {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-
-        } catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
