diff --git a/build/transport/WebSocketTransport.cjs b/build/transport/WebSocketTransport.cjs
index 602170a5e01ea5627b424663a0dee095289e3065..d02ae37facab51a8ee6ae11bcb7b38064ec74477 100644
--- a/build/transport/WebSocketTransport.cjs
+++ b/build/transport/WebSocketTransport.cjs
@@ -45,11 +45,13 @@ class WebSocketTransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     connect(url, headers) {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-        }
-        catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
diff --git a/build/transport/WebSocketTransport.mjs b/build/transport/WebSocketTransport.mjs
index 5b1d99fbc672c168a5123cc03c9f1aa2f367ea95..b63a0d705de4de784a4a14a6b7cac2a81fb4d9f6 100644
--- a/build/transport/WebSocketTransport.mjs
+++ b/build/transport/WebSocketTransport.mjs
@@ -46,11 +46,13 @@ class WebSocketTransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     connect(url, headers) {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-        }
-        catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
diff --git a/src/transport/WebSocketTransport.ts b/src/transport/WebSocketTransport.ts
index 92e86383eca4ec2b15ffb2ee406e9b41597a1b85..c499f0dbf391f6266ad667300ee825c525a151f7 100644
--- a/src/transport/WebSocketTransport.ts
+++ b/src/transport/WebSocketTransport.ts
@@ -48,11 +48,13 @@ export class WebSocketTransport implements ITransport {
      * @param headers custom headers to send with the connection (only supported in Node.js. Web Browsers do not allow setting custom headers)
      */
     public connect(url: string, headers?: any): void {
-        try {
+        const supportsNodeOptions = WebSocket === NodeWebSocket ||
+            (typeof process !== "undefined" &&
+                typeof process.versions?.node === "string");
+        if (supportsNodeOptions) {
             // Node or Bun environments (supports custom headers)
             this.ws = new WebSocket(url, { headers, protocols: this.protocols });
-
-        } catch (e) {
+        } else {
             // browser environment (custom headers not supported)
             this.ws = new WebSocket(url, this.protocols);
         }
